firmware/application/apps/ble_rx_app.cpp

```diff
@@ -428,42 +393,66 @@ void BLERxView::on_data(BlePacketData* packet) {
     // Start of Packet stuffing.
     // Masking off the top 2 bytes to avoid invalid keys.
     auto& entry = ::on_packet(recent, macAddressEncoded & 0xFFFFFFFFFFFF);
-    truncate_entries(recent, 32);
     updateEntry(packet, entry);
 
+    // Add entries if they meet the criteria.
+    auto value = filter;
+    resetFilteredEntries(recent, [&value](const BleRecentEntry& entry) {
+        return (entry.dataString.find(value) == std::string::npos) && (entry.nameString.find(value) == std::string::npos);
+    });
+
+    handle_entries_sort(options_sort.selected_index());
+
     // Log at End of Packet.
     if (logger && logging) {
         logger->log_raw_data(str_console);
     }
 }
 
 void BLERxView::on_switch_table(const std::string value) {
```

Comment:
nit: I don't know why the original authors did this, but making pass-by-value parameter const doesn't make sense really.
I delete that whenever I touch a file where it's happening.

Reply:
Like in the widget?

Reply:
Sorry I was looking at the wrong line. I think.